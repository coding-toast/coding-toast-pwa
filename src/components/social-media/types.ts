export interface SocialBrand {
  name: string
  logo: string
  href: string
}

export enum scale {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large'
}
