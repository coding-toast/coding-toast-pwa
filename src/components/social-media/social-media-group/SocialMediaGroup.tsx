import * as React from 'react'
import { Container, Row } from 'react-bootstrap'
import SocialBrandIcon from '../social-brand-icon/SocialBrandIcon'
import socialIcons from './SocialIcons'
import { scale, SocialBrand } from '../types'

interface SocialMediaGroupProps {
  iconScale?: scale
}

const scaleToEm = (iconScale: scale) => {
  switch (iconScale) {
    case 'small':
      return 2
    case 'medium':
      return 4
    case 'large':
      return 6
    default:
      return 4
  }
}

const SocialMediaGroup: React.FC<SocialMediaGroupProps> = (props) => {
  const { iconScale } = props
  const brandLogoEm = scaleToEm(iconScale == null ? scale.MEDIUM : iconScale)

  const brands = socialIcons.map((element: SocialBrand) => {
    return <SocialBrandIcon key={element.name} brand={element} size={brandLogoEm} />
  })

  return (
    <Container style={{ maxWidth: `${socialIcons.length * 2 + socialIcons.length * brandLogoEm}em` }} fluid>
      <Row className="justify-content-md-center">{brands}</Row>
    </Container>
  )
}

export default SocialMediaGroup
