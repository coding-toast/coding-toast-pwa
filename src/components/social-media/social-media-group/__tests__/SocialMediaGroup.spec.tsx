import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SocialMediaGroup from '../SocialMediaGroup'
import { scale } from '../../types'

describe('<SocialMediaGroup />', () => {
  it('renders correctly scale=small', () => {
    const component = shallow(<SocialMediaGroup iconScale={scale.SMALL} />)
    expect(toJson(component)).toMatchSnapshot()
  })
  it('renders correctly scale=medium', () => {
    const component = shallow(<SocialMediaGroup iconScale={scale.MEDIUM} />)
    expect(toJson(component)).toMatchSnapshot()
  })
  it('renders correctly scale=large', () => {
    const component = shallow(<SocialMediaGroup iconScale={scale.LARGE} />)
    expect(toJson(component)).toMatchSnapshot()
  })
  it('renders correctly with no scale modifier', () => {
    const component = shallow(<SocialMediaGroup />)
    expect(toJson(component)).toMatchSnapshot()
  })
})
