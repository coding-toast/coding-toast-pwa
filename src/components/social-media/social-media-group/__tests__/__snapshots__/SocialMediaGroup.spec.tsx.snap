// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialMediaGroup /> renders correctly scale=large 1`] = `
<Container
  fluid={true}
  style={
    Object {
      "maxWidth": "24em",
    }
  }
>
  <Row
    className="justify-content-md-center"
    noGutters={false}
  >
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "Twitter",
        }
      }
      key="Twitter"
      size={6}
    />
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "Reddit",
        }
      }
      key="Reddit"
      size={6}
    />
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "RSS",
        }
      }
      key="RSS"
      size={6}
    />
  </Row>
</Container>
`;

exports[`<SocialMediaGroup /> renders correctly scale=medium 1`] = `
<Container
  fluid={true}
  style={
    Object {
      "maxWidth": "18em",
    }
  }
>
  <Row
    className="justify-content-md-center"
    noGutters={false}
  >
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "Twitter",
        }
      }
      key="Twitter"
      size={4}
    />
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "Reddit",
        }
      }
      key="Reddit"
      size={4}
    />
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "RSS",
        }
      }
      key="RSS"
      size={4}
    />
  </Row>
</Container>
`;

exports[`<SocialMediaGroup /> renders correctly scale=small 1`] = `
<Container
  fluid={true}
  style={
    Object {
      "maxWidth": "12em",
    }
  }
>
  <Row
    className="justify-content-md-center"
    noGutters={false}
  >
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "Twitter",
        }
      }
      key="Twitter"
      size={2}
    />
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "Reddit",
        }
      }
      key="Reddit"
      size={2}
    />
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "RSS",
        }
      }
      key="RSS"
      size={2}
    />
  </Row>
</Container>
`;

exports[`<SocialMediaGroup /> renders correctly with no scale modifier 1`] = `
<Container
  fluid={true}
  style={
    Object {
      "maxWidth": "18em",
    }
  }
>
  <Row
    className="justify-content-md-center"
    noGutters={false}
  >
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "Twitter",
        }
      }
      key="Twitter"
      size={4}
    />
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "Reddit",
        }
      }
      key="Reddit"
      size={4}
    />
    <SocialBrandIcon
      brand={
        Object {
          "href": "/",
          "logo": "test-file-stub",
          "name": "RSS",
        }
      }
      key="RSS"
      size={4}
    />
  </Row>
</Container>
`;
